File: 0624/solution.py
class Solution:
    def maxDistance(self, arrays: list[list[int]]) -> int:
        small, big = arrays[0][0], arrays[0][-1]
        res = 0
        for i in range(1, len(arrays)):
            currs, currb = arrays[i][0], arrays[i][-1]
            res = max(res, currb - small, big - currs)
            small, big = min(small, currs), max(big, currb)
        return res

File: 0624/solution.ts
function maxDistance(arrays: number[][]): number {
  let res = 0;
  let prevBig = arrays[0][arrays[0].length - 1];
  let prevSmall = arrays[0][0];

  for (let i = 1; i < arrays.length; i++) {
    let currBig = arrays[i][arrays[i].length - 1];
    let currSmall = arrays[i][0];
    res = Math.max(res, currBig - prevSmall);
    res = Math.max(res, prevBig - currSmall);
    prevBig = Math.max(prevBig, currBig);
    prevSmall = Math.min(prevSmall, currSmall);
  }
  return res;
}
